### DISCUSSION TITLE
LocalPeopleResults search with default values doesn't display UserDisplayName (PersonImmersive.aspx instead)

### DISCUSSION THREAD

**Discussion by PierreFE** [microsoft-search/pnp-modern-search#2509] [Q&A]
Hi, 
I just try to use the pnp modern search to build our People Directory.
I try 4.8 after 4.7 to fix but it's not working better.
With both Chrome and Edge.

I let all the default values in search results :
![1](https://user-images.githubusercontent.com/31345906/199677057-99df4ef1-8127-4fa3-8450-881a5859aab4.png)

But UserDisplayNames don't display and "PersonImmersive.aspx" display instead.

I can display The Last Name if I map UserDisplayName on LastName but don't have First Name.

Anything wrong  ?

**Comment by wobba** [comment]
Sorry, but I don't understand what you are asking. Can you please elaborate?

**Reply by Lotsamocha01** [reply]
I believe I am also having this issue. Using the default People layout under the Search Results webpart displays some results but the Display Name is showing as "PersonImmersive.aspx" and the E-mail address is missing the last letter. When inspecting the image, the referenced e-mail address in the URL is also missing the last character. The domain is "@kbc.global" and the 'l' is missing on the end.

![image](https://user-images.githubusercontent.com/114703957/233742182-5dc0ca3e-906f-4e98-905e-6e70d94f60fd.png)

Image Element:

`<img class="ms-Image-image is-loaded ms-Image-image--cover ms-Image-image--landscape image-405" src="/_layouts/15/userphoto.aspx?size=L&amp;username=xxxxxxx@kbc.globa" alt="">`

**Comment by kasperbolarsen** [comment]
Hi,  can you please verify the data returned from the Search API using either SP Editor or the SP Search query tool. 
If you are not familiar with those tools, please see https://ms365thinking.blogspot.com/2023/04/new-videos-on-youtube-useful-tools-when.html?m=1

**Reply by Lotsamocha01** [reply]
Using the SP Search Query Tool, I can verify that the query results are displaying the correct data for the accounts. Although, it is still showing the "PersonImmersive.aspx" instead of the name.
![image](https://user-images.githubusercontent.com/114703957/234054664-709c8d33-1513-42d2-920a-6d33e11e7325.png)
![image](https://user-images.githubusercontent.com/114703957/234055055-5e94c75f-26de-4acc-8d99-b10fd38c7dee.png)
![image](https://user-images.githubusercontent.com/114703957/234055268-80bf0c16-697e-48e3-8efd-fcf554e821b8.png)

![image](https://user-images.githubusercontent.com/114703957/234055482-92e63590-7928-403e-9cee-05c2f99cd567.png)

**Reply by Lotsamocha01** [reply]
Looking at the Regex expression in "getUserEmail" I can see that it is limiting the top-level domain to 5 characters. As our domain has 6 characters, would this cause the issue?

[pnp-modern-search](https://github.com/microsoft-search/pnp-modern-search/tree/04f489ebbbb70b8096b5429d756f3fef39bf10c3)/[search-parts](https://github.com/microsoft-search/pnp-modern-search/tree/04f489ebbbb70b8096b5429d756f3fef39bf10c3/search-parts)/[src](https://github.com/microsoft-search/pnp-modern-search/tree/04f489ebbbb70b8096b5429d756f3fef39bf10c3/search-parts/src)/[services](https://github.com/microsoft-search/pnp-modern-search/tree/04f489ebbbb70b8096b5429d756f3fef39bf10c3/search-parts/src/services)/[templateService](https://github.com/microsoft-search/pnp-modern-search/tree/04f489ebbbb70b8096b5429d756f3fef39bf10c3/search-parts/src/services/templateService)/TemplateService.ts

![image](https://user-images.githubusercontent.com/114703957/234710168-2489ecec-acb2-4b78-9f8a-f9c75999aff7.png)

**Reply by wobba** [reply]
Tracked in #2933

**Reply by Lotsamocha01** [reply]
This doesn't resolve the "PersonImmersive.aspx" showing instead of the user name. I've worked around this by adding the "PreferredName" property and assigning it to the People Field for the Primary Text. Any ideas as to where we need to look for the reason this isn't showing correctly? Is this possibly an Office 365 issue?

![image](https://github.com/microsoft-search/pnp-modern-search/assets/114703957/01c98a2b-6624-4b34-a5f7-93b010e5a68c)

**Reply by kasperbolarsen** [reply]
Setting up a new results web part: 
Data source: SharePoint
Query Template : {searchTerms} *
Result SourceID: LocalPeopleResults
Layouts: People

gives
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/5996522b-1547-4de9-938b-1bb3809d865a)


and the "Manage People fields" are:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/cd270270-0ef4-4365-b027-228d748a0f2c)

and the layout slots looks like this:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/28f36c8a-d79e-422a-9fc0-3672ec24cdfe)

My mapping for the Title property is 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/ee9a6833-8524-4fd6-b96e-31346afb75af)

Can you spot any values in your setup that doesn't match mine ?

**Reply by Lotsamocha01** [reply]
Setting up a new results web part:
Data source: SharePoint
Query Template : {searchTerms} *
Result SourceID: LocalPeopleResults
Layouts: People

gives
![image](https://github.com/microsoft-search/pnp-modern-search/assets/114703957/f2e0c3e1-6761-48f6-86f3-b8814427d4c6)

Manage People Fields
![image](https://github.com/microsoft-search/pnp-modern-search/assets/114703957/7c5b5201-34e0-4b60-971d-2c5d7c73c49a)

Layout Slots
![image](https://github.com/microsoft-search/pnp-modern-search/assets/114703957/0d32008f-2083-4c1b-a51f-c57950c6689a)

Mapping for the Title property
![image](https://github.com/microsoft-search/pnp-modern-search/assets/114703957/a4667c5f-59a4-4029-98c6-c150a7504e19)

**Comment by kasperbolarsen** [comment]
OK, it looks like a bad case of "WrongDataInTheIndex"-titis :-( 
I would try to reindex the LocalPeopleResults by running the script https://pnp.github.io/script-samples/spo-request-pnp-reindex-user-profile/README.html?tabs=pnpps

**Reply by Lotsamocha01** [reply]
Does this reindex all of the non user editable fields or do I need to edit the script to specify the fields I need updating?

**Reply by kasperbolarsen** [reply]
The script will,  as far as I remember,  just cause all UPAs to be reindexed,  Inc all properties.

**Reply by Lotsamocha01** [reply]
It's showing as successful. Guess I'll wait a day and check to see if the index has been updated.

![image](https://github.com/microsoft-search/pnp-modern-search/assets/114703957/2aae06fc-451d-4e4c-b609-e8b9b49f5e21)

**Reply by Lotsamocha01** [reply]
PersonImmersive.aspx is still showing up in the results after 24 hours. I'll check again after the weekend.

**Reply by wobba** [reply]
Seems to be an API change from the Microsoft side. You can probably change to use First name and Lastname instead as a workaround.

**Reply by kasperbolarsen** [reply]
> Seems to be an API change from the Microsoft side. You can probably change to use First name and Lastname instead as a workaround.

Are you talking about an unannounced API , aka a bug?

I have looked for any announcement that *could* be related to this issue,  but found none.

@Lotsamocha01 if the proper data doesn't show up by Monday I will recommend a support ticket with Microsoft and report any findings. I know for a fact that I will be very busy rewriting a lot of code if our customers tenants starts returning gibberish in that property 😮‍💨

**Reply by wobba** [reply]
Bug seems appropriate indeed.

**Reply by wobba** [reply]
I cannot repro this myself in any tenants. Is it still an issue?

**Reply by Lotsamocha01** [reply]
I'm still seeing the issue on my tenant. I will need to open a ticket with Microsoft. 

![image](https://github.com/microsoft-search/pnp-modern-search/assets/114703957/3b60e2b8-2b07-4a18-bd06-b9ef76f34631)
